### PR TITLE
F2 f 133 framework fixes

### PR DESCRIPTION
**Pull request by merlinc**
### What changed

Refactor to use features of hmpo-form-wizard, including editing, backLinks, and config based routing.

Note: This should be backed up with browser tests covering all the scenarios required.

### Issue tracking

 
- [F2F-XXXX](https://govukverify.atlassian.net/browse/F2F-XXX)

### Environment variables or secrets

- [ ] Documented in the README
- [ ] Added screenshots to show the implementation is working
- [ ] Ran cfn-lint on any SAM templates

### CODE REVIEW

File: src/app/cic/controllers/brpDetails.js [hunk 1]
```diff
@@ -4,39 +4,5 @@ const DateControllerMixin = require("hmpo-components").mixins.Date;
 
 const DateController = DateControllerMixin(BaseController);
 
-class BrpDetailsController extends DateController {
-
-  async saveValues(req, res, next) {
```
Comment (merlinc) [inline src/app/cic/controllers/brpDetails.js:9]: No need for custom `saveValues` if the routing is calculated correctly.

Also, this function isn't calling the [super.saveValues](https://github.com/HMPO/hmpo-form-wizard/blob/master/lib/controller/index.js#L350), so wasn't saving the values correctly, which is where the extra `sessionModel.set` were coming from.

File: src/app/cic/controllers/checkDetails.js [hunk 2]
```diff
@@ -25,10 +24,6 @@ class CheckDetailsController extends DateController {
       callback(err, locals);
     });
   }
-
-  next() {
```
Comment (merlinc) [inline src/app/cic/controllers/checkDetails.js:29]: No need for functions as [next can be a relative string path](https://github.com/HMPO/hmpo-form-wizard/tree/master#next-steps)

File: src/app/cic/steps.js [hunk 3]
```diff
@@ -24,80 +24,219 @@ module.exports = {
...
   },
   "/landingPage": {
-
     next: "photoIdSelection",
   },
   "/photoIdSelection": {
     controller: photoIdSelect,
-    fields: ['photoIdChoice'],
-    next: photoIdSelect.prototype.next
+    editable: true,
```
Comment (merlinc) [inline src/app/cic/steps.js:31]: Set `editable` to be true, to allow the `/photoIdSelection/edit` link to work

File: src/app/cic/steps.js [hunk 4]
```diff
@@ -24,80 +24,219 @@ module.exports = {
...
+    editBackStep: "checkDetails",
```
Comment (merlinc) [inline src/app/cic/steps.js:32]: Set the `editBackStep` to be the details screen so backlinks work correctly.

**Comment by richa-misra** [comment]
changes incorporated in other PRs

**Pull request closed without merging**